src/components/lyne-footer/lyne-footer.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Prop
+} from '@stencil/core';
+import { InterfaceFooterAttributes } from './lyne-footer.custom';
+
+@Component({
+  shadow: true,
+  styleUrls: {
+    default: 'styles/lyne-footer.default.scss',
+    shared: 'styles/lyne-footer.shared.scss'
+  },
+  tag: 'lyne-footer'
+})
+
+export class LyneFooter {
+
+  /** Footer appearance */
+  @Prop() public appearance?: InterfaceFooterAttributes['appearance'] = 'primary';
+
+  /** Footer title text, visually hidden,  necessary for screenreaders */
+  @Prop() public accessibilityTitle!: string;
+
+  /** Host element */
+  @Element() private _hostElement: HTMLElement;
+
+  private _hasCol1Slot: boolean;
+  private _hasCol2Slot: boolean;
+  private _hasCol3Slot: boolean;
+  private _hasCol4Slot: boolean;
+  private _hasClockSlot: boolean;
+  private _hasBottomSlot: boolean;
+
+  public componentWillLoad(): void {
+    // Check slots
+    this._hasCol1Slot = Boolean(this._hostElement.querySelector('[slot="col-1"]'));
+    this._hasCol2Slot = Boolean(this._hostElement.querySelector('[slot="col-2"]'));
+    this._hasCol3Slot = Boolean(this._hostElement.querySelector('[slot="col-3"]'));
+    this._hasCol4Slot = Boolean(this._hostElement.querySelector('[slot="col-4"]'));
+    this._hasClockSlot = Boolean(this._hostElement.querySelector('[slot="clock"]'));
+    this._hasBottomSlot = Boolean(this._hostElement.querySelector('[slot="bottom"]'));
+  }
+
+  public render(): JSX.Element {
+
+    const className = `footer footer--${this.appearance}`;
+
+    const attrs = {
+      class: className
+    };
+
+    return (
+      <Host role='contentinfo'>
```

Comment:
Ok, my take here was to define it as far outside as possible and assumed this would help the user the most since it is rendered on the host element. But I moved it inside.